components/brave_search/browser/brave_search_default_host_unittest.cc

```diff
@@ -157,9 +152,7 @@ TEST_F(BraveSearchDefaultHostTest, DisallowsAfterMaxTimesAsked) {
 TEST_F(BraveSearchDefaultHostTest, CanSetDefaultAlwaysTestWithSearchPromotion) {
   base::test::ScopedFeatureList feature_list;
 
-  NiceMock<LocaleHelperMock> locale_helper_mock;
-  brave_l10n::LocaleHelper::GetInstance()->SetForTesting(&locale_helper_mock);
-  ON_CALL(locale_helper_mock, GetLocale()).WillByDefault(Return("en-US"));
+  brave_l10n::SetLocaleForTesting("en_US");
```

Comment:
`brave_l10n::LocaleHelperMock` was used in many tests to restore the original locale after tests completed. We need to keep this logic, otherwise we will randomly break other tests that depend on the original locale. Please replace `SetLocaleForTesting` with a scoped helper.

Reply:
Fixed